syntax = "proto3";

import "streamlit/proto/LabelVisibilityMessage.proto";

message NumberInput {
  enum DataType {
    // Does the input operate on ints or floats? This doesn't change how the
    // data is stored, but the frontend needs to know for input parsing.
    INT = 0;
    FLOAT = 1;
  }

  string id = 1;
  string label = 2;
  string form_id = 3;

  string format = 8;
  bool has_min = 11;
  bool has_max = 12;

  DataType data_type = 13;
  double default = 14;
  double step = 15;
  double min = 16;
  double max = 17;
  string help = 18;
  double value = 19;
  bool set_value = 20;
  bool disabled = 21;
  LabelVisibilityMessage label_visibility = 22;

  reserved 4, 5, 6, 7, 9, 10;
}
